dev-haskell/hsopenssl/hsopenssl-0.11.7.9.ebuild

```
@@ -1,16 +1,21 @@
-# Copyright 1999-2023 Gentoo Authors
+# Copyright 1999-2025 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
 EAPI=8
 
-# ebuild generated by hackport 0.8.0.0.9999
+# ebuild generated by hackport 0.9.0.0.9999
 #hackport: flags: -homebrew-openssl,-macports-openssl,-use-pkg-config
 
 CABAL_PN="HsOpenSSL"
 
 CABAL_FEATURES="lib profile haddock hoogle hscolour test-suite"
 inherit haskell-cabal
 
+#Allow to compile with openssl>=3.0
+# see https://github.com/haskell-cryptography/HsOpenSSL/issues/93
+# and https://src.fedoraproject.org/rpms/ghc-HsOpenSSL/c/a3062db8017ceb96ccd18b915aa8e017c42b3a22?branch=rawhide
+HCFLAGS="-optc=-Wno-incompatible-pointer-types"
```

I'm not sure that this should be global based on other ebuilds which modify HCFLAGS (see `protocol-buffers-descriptors` for an example which places something like this in `src_prepare()`).

Plus this seems to overwrite whatever is in HCFLAGS rather than appending to the list of HCFLAGS, which I am not sure is the intended behaviour either.

---

I don't think it's necessary to change this in `src_prepare()`, but it's probably a good idea to make sure it doesn't clobber existing `HCFLAGS` that might have been passed from the command line or `make.conf`.

```suggestion
HCFLAGS+=" -optc=-Wno-incompatible-pointer-types"
```

---

The real question is if this workaround is a good idea

(https://wiki.gentoo.org/wiki/Modern_C_porting#-Wincompatible-pointer-types)

---

I suspect we may need to leave out this HCFLAGS change if hololeap determines it is inappropriate, until a better solution arrives. Fedora have done it, but that doesn't mean it's necessarily right for Gentoo.

---

The only other solution that comes to mind is to last-rite `HSOpenSSL`. For instance, here is a discussion in another project about dropping `snap` due to it depending on `HSOpenSSL`:

https://github.com/input-output-hk/iohk-monitoring-framework/pull/650

Also, from the wiki page I linked earlier, it doesn't sound like something we'll be able to patch easily:

> - These bugs are the hardest to solve and often require understanding the intent of the software's author. It's okay to feel stuck with these.
> - Many of these end up being last-rite candidates because they're abandoned upstream and have other code smells.

---

`HSOpenSSL` has a good few revdeps is all. It's not abandoned upstream either. I don't understand enough about the `HCFLAGS` change to comment on that. I wonder if it may be worth asking sam_ to weigh in in between his 40,000 concurrent tasks he's probably dealing with in Gentoo.

---

I opened a dedicated bug about this: https://github.com/gentoo-haskell/gentoo-haskell/issues/1644

---

I agree, let's last-rite and remove until upstream catches up.

---

@GuillaumedeVolpiano I suggest that you may drop this HSOpenSSL stuff from your PR now. It is now masked until upstream catches up. We can deal with dep versions at that point in time.

---

@solpeth done